pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -438,3 +473,85 @@ def test_unit_integrity(self):
         for i, module in enumerate(self.rpms[1:]):
             with self.subTest(i=i):
                 self.assertEqual(self.rpms[0], module)
+
+
+class SyncOnDemandTestCase(_BaseTestCase):
+    """Assert Yum supports on-demand syncing."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an RPM repository with a valid feed and sync it.
+
+        Do the following:
+
+        0. Reset Pulp, including the Squid cache.
+        1. Sync/publish a repository using the 'on_demand' download policy.
+        2. Download an RPM from the published repository.
+        3. Download the same RPM to ensure it is served by the cache.
+        """
+        super(SyncOnDemandTestCase, cls).setUpClass()
+
+        # Required to ensure the `locally_stored_units` is 0 before we start.
+        utils.reset_squid(cls.cfg)
+        utils.reset_pulp(cls.cfg)
+        importer_config = {
+            'feed': RPM_FEED_URL,
+            'download_policy': 'on_demand',
+        }
+        repo, tasks = _sync_and_publish_repo(
+            cls.cfg,
+            importer=importer_config
+        )
+        client = api.Client(cls.cfg, api.echo_handler)
+        cls.repo = client.get(repo['_href'], params={'details': True}).json()
+
+        url = urljoin(
+            '/pulp/repos/',
+            repo['id'] + '/',
+        )
+        url = urljoin(url, RPM)
+        cls.rpm = client.get(url)
+        cls.same_rpm = client.get(url)
+
+    def test_local_units(self):
+        """Assert no content units were downloaded besides metadata."""
+        metadata_units = filter(
+            lambda unit_tuple: unit_tuple[0] not in ('rpm', 'drpm', 'srpm'),
+            self.repo['content_unit_counts'].items()
+        )
+        metadata_unit_count = sum([unit[1] for unit in metadata_units])
```

Comment:
Is `repo['content_unit_counts']` a dict in the following form?

```python
{
    'foo': 1,
    'bar': 3,
    'biz': 10,
    'baz': 5,
}
```

If so, this can be written as a list comprehension:

```python
>>> counts = {'foo': 1, 'bar': 2, 'biz': 3, 'baz': 4}
>>> sum((
...     number for name, number in counts.items()
...     if name not in {'foo', 'bar'}
... ))
7
```